SpatialGDK/Source/SpatialGDK/Private/Utils/Interest/ResultTypes.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/Interest/ResultTypes.h"
+
+#include "UObject/UObjectIterator.h"
+
+#include "SpatialGDKSettings.h"
+
+namespace SpatialGDK
+{
+
+ResultType ResultTypes::CreateClientNonAuthInterestResultType(USpatialClassInfoManager* ClassInfoManager)
+{
+	ResultType ClientNonAuthResultType;
+
+	// Add the required unreal components
+	ClientNonAuthResultType.Append(SpatialConstants::REQUIRED_COMPONENTS_FOR_NON_AUTH_CLIENT_INTEREST);
+
+	// Add all data components- clients don't need to see handover or owner only components on other entities.
+	ClientNonAuthResultType.Append(ClassInfoManager->GetComponentIdsForComponentType(ESchemaComponentType::SCHEMA_Data));
+
+	// In direct disagreement with the above comment, we add the owner only components as well.
+	// This is because GDK workers currently make assumptions about information being available at the point of possession.
+	// TODO(jacques): fix (unr-2865)
+	ClientNonAuthResultType.Append(ClassInfoManager->GetComponentIdsForComponentType(ESchemaComponentType::SCHEMA_OwnerOnly));
+
+	return ClientNonAuthResultType;
+}
+
+ResultType ResultTypes::CreateClientAuthInterestResultType(USpatialClassInfoManager* ClassInfoManager)
+{
+	ResultType ClientAuthResultType;
+
+	// Add the required known components
+	ClientAuthResultType.Append(SpatialConstants::REQUIRED_COMPONENTS_FOR_AUTH_CLIENT_INTEREST);
+	ClientAuthResultType.Append(SpatialConstants::REQUIRED_COMPONENTS_FOR_NON_AUTH_CLIENT_INTEREST);
+
+	// Add all the generated unreal components
+	ClientAuthResultType.Append(ClassInfoManager->GetComponentIdsForComponentType(ESchemaComponentType::SCHEMA_Data));
+	ClientAuthResultType.Append(ClassInfoManager->GetComponentIdsForComponentType(ESchemaComponentType::SCHEMA_OwnerOnly));
+
+	return ClientAuthResultType;
+}
+
+ResultType ResultTypes::CreateServerNonAuthInterestResultType(USpatialClassInfoManager* ClassInfoManager)
+{
+	ResultType ServerNonAuthResultType;
+
+	// Add the required unreal components
+	ServerNonAuthResultType.Append(SpatialConstants::REQUIRED_COMPONENTS_FOR_NON_AUTH_SERVER_INTEREST);
+
+	// Add all data, owner only, and handover components
+	ServerNonAuthResultType.Append(ClassInfoManager->GetComponentIdsForComponentType(ESchemaComponentType::SCHEMA_Data));
```

Comment:
There's a lot of duplication between these functions. I feel like there could be a general helper function where you pass it some SpatialConstants, and ESchemaComponentType enums, and it fills out the ResultType for you, and then these functions make use of that helper.

Reply:
Wouldn't that be roughly the same, but instead of using `Append` you call a helper function? 

I was considering making a `ResultTypesBuilder` instead of appending to the array that would do something similar to this, however I think this approach seemed nicer in the end as each line would still be present in both approaches and the added complexity of a builder I don't think is required as the array can't be accessed until the function is finished

Reply:
Fair enough. My main point is that the internal of this functions should mostly be a list of enums/constants but I guess it basically is.

Reply:
> Wouldn't that be roughly the same, but instead of using Append you call a helper function?
Yes, but it would be nicer to read `AddComponent(s)ToResultType` over `Append`, but it is a minor point.

Reply:
Spoke offline- leaving this for now as I don't think the added complexity of a helper is worth the readability in this case.